components/content_settings/core/browser/brave_content_settings_utils.cc

```diff
@@ -21,6 +23,32 @@ const std::vector<std::string> kShieldsResourceIDs {
     brave_shields::kReferrers,
     brave_shields::kCookies };
 
+bool CanPatternBeConvertedToWildcardSchemeAndPort(
+    const ContentSettingsPattern& pattern) {
+  // 1. Wildcard is alerady in the desired state.
+  // 2. Our firstParty placeholder shouldn't be converted.
+  // 3. Patterns that have file:// scheme.
+  // 4. We only want to convert patterns that have a specific host, so something
+  // like "http://*:80/*" should be left alone.
+  if (pattern == ContentSettingsPattern::Wildcard() ||
+      pattern == ContentSettingsPattern::FromString("https://firstParty/*") ||
+      pattern.GetScheme() == ContentSettingsPattern::SCHEME_FILE ||
```

Comment:
what is special about FILE? Do they have port at all?

Reply:
I don't think Shields are active on a file:// URLs, and you can't add one via Chromium's settings UI, so I don't think we'd encounter this, but just as a precaution there is some strangeness with file:// URLs. `ContentSettingsPattern::FromString( "file://brave.com:8080/a/b/c.zip")` for example would return an empty ContentSettingsPattern, while `ContentSettingsPattern::FromURL( GURL("file://brave.com:8080/a/b/c.zip"))` would return essentially a `file:///a/b/c.zip` pattern where host is empty and is not a wildcard, so it wouldn't get rejected by `pattern.MatchesAllHosts()`. Perhaps this function should also check if the host is empty.

Reply:
See changes in bdc2851b5386b1ae42a134550a217edb6267ef57

Reply:
sounds good